pyccel/codegen/compiling/basic.py

```diff
@@ -0,0 +1,237 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details.     #
+#------------------------------------------------------------------------------------------#
+"""
+Module handling classes for compiler information relevant to a given object
+"""
+import os
+import sys
+from filelock import FileLock
+
+class CompileObj:
+    """
+    Class containing all information necessary for compiling
+
+    Parameters
+    ----------
+    file_name     : str
+                    Name of file to be compiled
+
+    folder        : str
+                    Name of the folder where the file is found
+
+    is_module     : bool
+                    Indicates whether we are compiling a module or a program
+                    Default : True
+
+    flags         : str
+                    Any non-default flags passed to the compiler
+
+    includes      : iterable of strs
+                    include directories paths
+
+    libs          : iterable of strs
+                    required libraries
+
+    libdirs       : iterable of strs
+                    paths to directories containing the required libraries
+
+    dependencies  : iterable of CompileObjs
+                    objects which must also be compiled in order to compile this module/program
+
+    accelerators  : str
+                    Tool used to accelerate the code (e.g. openmp openacc)
+    """
+    def __init__(self,
+                 file_name,
+                 folder,
+                 is_module    = True,
+                 flags        = (),
+                 includes     = (),
+                 libs         = (),
+                 libdirs      = (),
+                 dependencies = (),
+                 accelerators = ()):
+        if not all(isinstance(d, CompileObj) for d in dependencies):
+            raise TypeError("Dependencies require necessary compile information")
+
+        self._file = os.path.join(folder, file_name)
+        self._folder = folder
+
+        self._module_name = os.path.splitext(file_name)[0]
+        rel_mod_name = os.path.join(folder, self._module_name)
+        self._module_target = rel_mod_name+'.o'
+
+        self._prog_target = rel_mod_name
+        if sys.platform == "win32":
+            self._prog_target += '.exe'
+
+        self._target = self._module_target if is_module else self._prog_target
+        self._target = os.path.join(folder, self._target)
+
+        self._lock         = FileLock(self.target+'.lock')
+
+        self._flags        = list(flags or ())
```

Comment:
the default value for flags is seted as `()` so it is enough to write just `self._flags        = list(flags)`

Reply:
The problem is that the default value [here](https://github.com/pyccel/pyccel/blob/master/pyccel/codegen/pipeline.py#L48-L64) is `None`, so that gets passed through anyway, hence the problem here. Do you think I should change the defaults here? I would prefer to change the defaults in `execute_pyccel` but I'm not sure if it will work. I will try, I might have changed enough for it to be ok now

Reply:
My bad, you are right that works now. Fixed in 518629a